tailormap-components/projects/core/src/lib/feature-form/state/form.state-helpers.ts

```diff
@@ -31,3 +23,16 @@ export const updateFeatureInArray = (features: Feature[], newFeature: Feature):
       children: feature.children ? updateFeatureInArray(feature.children, newFeature) : null,
     }));
 };
+
+export const addFeatureToParent = (features: Feature[], newFeature: Feature, parentId:string) : Feature[] =>{
+  const idx = features.findIndex(feature => feature.objectGuid === parentId);
+
+  return (idx !== -1 ?
```

Comment:
Since its already a somewhat complicated statement you could also do an early return:

```
if (idx === -1) {
    return features;
}
return {
   ...features.slice etc
```